storage/src/test/scala/coop/rchain/storage/LmdbTests.scala

```diff
@@ -7,132 +7,54 @@
 
 package coop.rchain.storage
 
+import java.io.File
 import java.lang._
+import org.scalatest._
 
 // TODO: add timings to tests (get and put)
 
-object LmdbTests {
-
-  def tests(): Unit = {
-
-    println("LmdbTests.tests() begin")
-
-    updateIntInt()
-    updateIntInts()
-
-    putGetDupSort()
-
-    iterateStrStrs()
-    iterateIntInts()
-
-    intInt()
-    intInts()
-
-    intStr()
-    intStrs()
-
-    intIntDelete()
-    intIntsDelete()
-
-    stringLongsDelete()
-    longFloatsDelete()
-    floatDoublesDelete()
-
-    strLongs()
-    intFloats()
-    floatDoubles()
-
-    // TODO: test read only db
-    // TODO: test big blob value
-    // TODO: pass cursors in to methods
-
-    println("LmdbTests.tests() end")
-  }
-
-  def putGetDupSort(): Unit = {
-
-    println("putGetDupSort() begin")
+class LmdbTests extends FlatSpec with Matchers {
+    val basePath = System.getProperty("user.dir") + "/"
 
+  "Lmdb with keys that are ints and strings are associated with multiple values" should "retrieve the expected values" in {
     val dirName: Option[String] = Some("lmdbPutGetDupSortDb")
     val name: Option[String] = Some("lmdbPutGetDupSort")
 
     val lmdb = new Lmdb(dirName, name, true, true)
 
-    lmdb.put(1, 2)
-    val getInt = lmdb.getInts(1)
-    assert(getInt.get(0) == 2)
-
-    lmdb.put(3, 4)
-    lmdb.put(3, 5)
-    val getInts = lmdb.getInts(3)
-    assert(getInts.get(0) == 4)
-    assert(getInts.get(1) == 5)
-
-    lmdb.put(10L, 20L)
-    // val getInt = lmdb.getLongLongs(10L)
-    // assert(getInt.get(0) == 2)
-
-    lmdb.put(3, 4)
-    lmdb.put(3, 5)
-    // val getInts = lmdb.getIntInts(3)
-    // assert(getInts.get(0) == 4)
-    // assert(getInts.get(1) == 5)
-
-    lmdb.put("a", "b")
-    val getStr = lmdb.getStrings("a")
-    val str = getStr.get(0)
-    assert(str == "b")
-
-    lmdb.put("c", "d")
-    lmdb.put("c", "e")
-    val getStrings = lmdb.getStrings("c")
-    assert(getStrings.get(0) == "d")
-    assert(getStrings.get(1) == "e")
-
-    lmdb.deleteFiles()
-
-    println("putGetDupSort() end")
-  }
-
-  def iterateIntInts(): Unit = {
-
-    println("iterateIntIntsDb() begin")
-
-    val dirName: Option[String] = Some("lmdbIterateIntIntsDb")
-    val name: Option[String] = Some("lmdbIterateIntInts")
-
-    val numKeys = 3
-    val valuesCount = 2
-
-    val lmdb = new Lmdb(dirName, name, true, true)
-
     try {
-      for (i <- 0 until numKeys) {
-        print(s"put($i): ")
-        for (v <- 0 until valuesCount) {
-          print((i + 1 + v) + ", ")
-          lmdb.put(i, i + 1 + v)
-        }
-        println()
-      }
-
-      lmdb.displayRowsIntInt()
+      lmdb.put(1, 2)
+      val getInt = lmdb.getInts(1)
+      assert(getInt.get(0) == 2)
+
+      lmdb.put(3, 4)
+      lmdb.put(3, 5)
+      val getInts = lmdb.getInts(3)
+      assert(getInts.get(0) == 4)
+      assert(getInts.get(1) == 5)
```

Comment:
Is a conformant implementation of storage required to preserve order?

Reply:
No, that's not a requirement.  Good point about that test.